pilot/pkg/xds/delta.go

```diff
@@ -393,21 +393,22 @@ func (s *DiscoveryServer) shouldRespondDelta(con *Connection, request *discovery
 	previousInfo.AlwaysRespond = false
 	con.proxy.Unlock()
 
-	oldAck := slices.EqualUnordered(previousResources, currentResources)
+	unchanged := slices.EqualUnordered(previousResources, currentResources)
 	// Spontaneous DeltaDiscoveryRequests from the client.
 	// This can be done to dynamically add or remove elements from the tracked resource_names set.
 	// In this case response_nonce is empty.
-	newAck := request.ResponseNonce != ""
-	if newAck != oldAck {
+	spontaneousReq := request.ResponseNonce == ""
+	if spontaneousReq && unchanged || !spontaneousReq && !unchanged {
 		// Not sure which is better, lets just log if they don't match for now and compare.
-		deltaLog.Errorf("ADS:%s: New ACK and old ACK check mismatch: %v vs %v", stype, newAck, oldAck)
+		deltaLog.Errorf("ADS:%s: Spontaneous request and old ACK check mismatch: %v vs %v", stype, spontaneousReq, unchanged)
```

Comment:
This is misleading when it enters this condion because of !spontaneousReq request check

Reply:
@ramaraochavali What do you feel now